src/client/power-pages/create/utils/CommonUtils.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { exec } from "child_process";
+import { existsSync, stat } from "fs";
+import path from "path";
+import * as vscode from "vscode";
+import {
+    NOT_A_PORTAL_DIRECTORY,
+    WEBSITE_YML,
+} from "../CreateOperationConstants";
+
+export const isNullOrEmpty = (str: string | undefined): boolean => {
+    return !str || str.trim().length === 0;
+};
+
+/**
+ * @example
+ * formatFolderName("My Folder Name"); // "my-folder-name"
+ */
+export const formatFolderName = (name: string) => {
+    return name.replace(/[/ ]/g, "-").toLowerCase();
+};
+
+/**
+ * @example
+ * formatFileName("My File Name"); // "My-File-Name"
+ */
+export const formatFileName = (name: string) => {
+    const words = name.trim().split(/[/ ]/);
+
+    // Uppercase the first letter of each word and join the words back together
+    return words.map((word) => word[0].toUpperCase() + word.slice(1)).join("-");
+};
+
+export function createFileWatcher(
+    context: vscode.ExtensionContext,
+    selectedWorkspaceFolder: string,
+    pattern: string
+) {
+    const watcher: vscode.FileSystemWatcher =
+        vscode.workspace.createFileSystemWatcher(
+            new vscode.RelativePattern(selectedWorkspaceFolder, pattern),
+            false,
+            true,
+            true
+        );
+
+    context.subscriptions.push(watcher);
+    return watcher;
+}
+
+export async function createRecord(
+    entityType: string,
+    execCommand: string,
+    portalDirectory: string,
+    watcher: vscode.FileSystemWatcher
+) {
+    await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: `Creating ${entityType}...`,
+            cancellable: false,
+        },
+        async (progress) => {
+            return new Promise<void>((resolve, reject) => {
+                watcher.onDidCreate(async (uri) => {
+                    // Stop watching the directory to avoid duplicate events
+                    watcher.dispose();
+                    try {
+                        await vscode.window.showTextDocument(uri);
+                        vscode.window.showInformationMessage(
+                            `${entityType} created!`
+                        );
+                        resolve();
+                    } catch (error) {
+                        reject(error);
+                    }
+                });
+                exec(execCommand, { cwd: portalDirectory }, (error) => {
+                    if (error) {
+                        vscode.window.showErrorMessage(error.message);
```

Comment:
The error.message here will not be localizable - lets log the error.message and show better generic message instead

Reply:
Take this change in next PR

Reply:
Sure, will take this up in followup PR: https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3226760